### ISSUE TITLE
[java] UnusedPrivateField and SingularField false positive with lombok annotation EqualsAndHashCode

### ISSUE THREAD

**Issue by berkam** [pmd/pmd#2673]
**Affects PMD Version:** 

6.26.0

**Rule:** [UnusedPrivateField](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedprivatefield)

**Description:**

I am using two objects of the same class for comparison. The class contains private fields and the lombok  annotation EqualsAndHashCode . This implementation causes an false positive response.

**Exception Stacktrace:**

```
.(TestClass.java:10)
 at line `8` and column `13`
Perhaps 'value' could be replaced by a local variable.
```

**Code Sample demonstrating the issue:**

```java
@EqualsAndHashCode
public class TestClass {

    private int value;

    public TestClass(int value) {
        this.value = value;
    }
}
```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

See code sample.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
